systeminit/si
# Ensure correct deletion order (ENG-1120,ENG-1200)

## Description

This PR introduces two bug fixes:

1. Ensure the order of recommendations (specifically "deletion" ones) is correct
2. Ensure `ConfirmationViews` are not created for `Components` that do not have "confirmations"

This PR also introduces slight refactors due to the increasingly complexity of the "confirmations" and "fix flow" paradigms, including a new submodule underneath the "confirmation" submodule.

## Commit Description

```
Primary:
- Add and track the "PrimaryActionKind" for set of ConfirmationViews for
  a given Component
- Sort all ConfirmationViews in the workspace by "PrimaryActionKind",
  which ensures the correct deletion order for nodes is in place
- Ensure ConfirmationViews aren't generated for Components that do not
  have confirmations

Secondary:
- Re-add "deletion" scenario to the "list_confirmations" integration
  test
- Replace integration test usages of "ActionKind::Other" with
  "ActionKind::Create" where appropriate

Misc:
- Delete unused ComponentError variants and re-order the remaining
  variants
- Fix clippy lint in config-file
- Ensure ChangeStatus can be copied
```

## GIF

<img src="https://media0.giphy.com/media/3o6nV8TW8AZOqkOHVm/giphy.gif"/>
